File: Backend/HAMSGateWay/Services/NotificationService.cs
using System;
using System.Net.Http;
using System.Net.Http.Json;
using System.Threading.Tasks;
using HAMSGateWay.DTOs;

namespace HAMSGateWay.Services
{
    public class NotificationService
    {
        private readonly string baseUrl = "https://localhost:7033/api/"; 

        public async Task<List<NotificationDTO>> GetUserNotifications(int userId)
        {
            List<NotificationDTO> notifications = new List<NotificationDTO>();
            using (var client = new HttpClient())
            {
                client.BaseAddress = new Uri(baseUrl);
                var response = await client.GetAsync($"Notification/user-notifications/{userId}");

                if (response.IsSuccessStatusCode)
                {
                    notifications = await response.Content.ReadFromJsonAsync<List<NotificationDTO>>();
                }
            }
            return notifications;
        }

        public async Task<List<NotificationDTO>> GetUpcomingReminders(int userId)
        {
            List<NotificationDTO> reminders = new List<NotificationDTO>();
            using (var client = new HttpClient())
            {
                client.BaseAddress = new Uri(baseUrl);
                var response = await client.GetAsync($"Notification/reminders/{userId}");
                Console.WriteLine($"{baseUrl}Notification/reminders/{userId}");

                if (response.IsSuccessStatusCode)
                {
                    reminders = await response.Content.ReadFromJsonAsync<List<NotificationDTO>>();
                }
            }
            return reminders;
        }

        public async Task<bool> MarkNotificationAsRead(int notificationId)
        {
            using (var client = new HttpClient())
            {
                client.BaseAddress = new Uri(baseUrl);
                var response = await client.PutAsJsonAsync($"Notification/mark-as-read/{notificationId}", notificationId);

                if (response.IsSuccessStatusCode)
                {
                    return true;
                }

                return false;
            }
        }
    }
}

File: Frontend/HAMS/src/app/services/notification.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { CreateNotificationDTO, NotificationDTO } from '../shared/models/notification.model';

@Injectable({
  providedIn: 'root'
})
export class NotificationService {

  private baseUrl = 'https://localhost:7078/api/Notification'; // Replace with your actual API URL

  constructor(private http: HttpClient) {}

  getUserNotifications(userId: number): Observable<NotificationDTO[]> {
    return this.http.get<NotificationDTO[]>(`${this.baseUrl}/user-notifications/${userId}`);
  }

  getUpcomingReminders(userId: number): Observable<NotificationDTO[]> {
    return this.http.get<NotificationDTO[]>(`${this.baseUrl}/reminders/${userId}`);
  }

  markNotificationAsRead(notificationId: number): Observable<void> {
    return this.http.put<void>(`${this.baseUrl}/mark-as-read/${notificationId}`, {});
  }
}
